From: Celeste <20312-Celeste@users.gitlab.alpinelinux.org>
Date: Sun, 05 May 2024 17:21:00 +0000
Subject: [PATCH] OpenJDK 21 compatibility

  - Update -source/-target to 8, as OpenJDK 21
    has removed support for 7.

  - Upgrade mvel to 2.5.2, which fixes the
    "java.lang.ClassNotFoundException: java.lang.Compiler" error
    due to OpenJDK 21 removing that class.

---

--- a/build.xml
+++ b/build.xml
@@ -25,8 +25,8 @@
   <property name="dist"  location="dist"/>
   <property name="lib" location="lib" />
 
-  <property name="javac.source" value="1.7" />
-  <property name="javac.target" value="1.7" />
+  <property name="javac.source" value="1.8" />
+  <property name="javac.target" value="1.8" />
 
   <property name="ivy.jar.version" value="2.5.0"/>
   <property name="ivy.jar.name" value="ivy-${ivy.jar.version}.jar"/>
@@ -131,7 +131,7 @@
   </target>
 
   <target name="generate-sources" depends="init">
-    <ivy:cachepath organisation="org.mvel" module="mvel2" revision="2.3.2.Final"
+    <ivy:cachepath organisation="org.mvel" module="mvel2" revision="2.5.2.Final"
       inline="true" conf="default" transitive="true" pathid="mvel.classpath"/>
     <mkdir dir="${build}/java" />
     <java
